### PR TITLE
Modify heat_2d tests to run as multi-locale perf tests

### PR DESCRIPTION
**Pull request by jeremiah-corrado**
The performance tests added in https://github.com/chapel-lang/chapel/pull/22655 were intended to be multi-locale tests but did not have the correct associated files to be included in our nightly multi-locale performance testing.

This PR updates them s.t. they can run nightly. The single-locale graph file was removed because the results from running these tests in a single-locale setting would not be particularly interesting.

- [x] passing w/ CHPL_COMM=gasnet and `-performance -perflabel ml-`

### CODE REVIEW

File: test/ML-GRAPHFILES [hunk 1]
```diff
@@ -156,3 +156,4 @@ performance/array/distCreate-large-deinit.ml-perf.graph
 library/packages/Sort/performance/dist-performance.ml-perf.graph
 runtime/ferguson/cache-remote-mock-tests/mock-ra-perf.ml-time.graph
 optimizations/autoAggregation/perf/autoAggNegativeImpact.ml-perf.graph
+exercises/diffusion/heat_2d_dist.ml-perf.graph
```
Comment (ronawho) [inline test/ML-GRAPHFILES:159]: `ml-perf.graph` -> `ml-time.graph`

Comment (jeremiah-corrado) [reply test/ML-GRAPHFILES:159]: oops, thanks!

**Review by ronawho — APPROVED** [review]

**Comment by bradcray** [comment]
@jeremiah-corrado : Sorry that I was too lazy to remind myself of the proper way to do this the first time / too forgetful to remember the difference between the two mechanisms.

**Comment by jeremiah-corrado** [comment]
No worries, I'm glad we discussed it in the meeting!

**Pull request merged**